addons/udes_stock/tests/test_picking_type.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+
+from . import common
+
+from odoo.exceptions import ValidationError
+
+
+class TestPickingType(common.BaseUDES):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestPickingType, cls).setUpClass()
+        cls.pack_4apples_info = [
+            {'product': cls.apple, 'qty': 4}
+        ]
+        cls.pack_4apples_3bananas_info = [
+            {'product': cls.apple, 'qty': 4},
+            {'product': cls.banana, 'qty': 3},
+        ]
+
+    def setUp(self):
+        super(TestPickingType, self).setUp()
+
+        Package = self.env['stock.quant.package']
+
+        self.package_one   = Package.get_package("test_package_one", create=True)
+        self.package_two   = Package.get_package("test_package_two", create=True)
+        self.package_three = Package.get_package("test_package_three", create=True)
+        self.package_four  = Package.get_package("test_package_four", create=True)
+
+    def test01_get_suggested_locations_exact_match(self):
+        """ Suggested location should exactly match with location_dest_id of the
+            stock.move.line
+        """
+        self.picking_type_putaway.u_drop_location_policy = 'exactly_match_move_line'
+        self.create_quant(self.apple.id,
+                          self.picking_type_putaway.default_location_src_id.id,
+                          4, package_id=self.package_one.id)
+        picking = self.create_picking(self.picking_type_putaway,
+                            products_info=self.pack_4apples_info,
+                            confirm=True,
+                            assign=True)
+
+        mls = picking.move_line_ids
+        mls.write({'location_dest_id': self.test_location_01.id})
+        location = picking.get_suggested_locations(move_line_ids=mls)
+        self.assertEqual(location, self.test_location_01)
+
+    def test02_get_suggested_locations_by_product(self):
+        """ Suggested locations should be all where the product is at.
```

Comment:
This should be rephrased to something like: "All of the suggested locations should have current stock of the product"